src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -349,35 +367,42 @@ static ucs_status_t uct_cuda_copy_mem_alloc(uct_md_h uct_md, size_t *length_p,
     alloc_handle->length = *length_p;
 
     if (mem_type == UCS_MEMORY_TYPE_CUDA) {
-
-        for (method = methods; method < methods + num_methods; ++method) {
+        ucs_carray_for_each(method, methods, num_methods) {
             switch (*method) {
-                case UCT_CUDA_ALLOC_METHOD_FABRIC_PINNED:
-                    status = uct_cuda_copy_mem_alloc_fabric(md, alloc_handle);
-                    if (status != UCS_OK) {
-                        continue;
-                    }
+            case UCT_CUDA_ALLOC_METHOD_FABRIC_DEVICE:
+                if (!md->config.fabric_supported) {
```

Comment:
can just skip to next method if fabric is not supported. otherwise there will be error on old devices that don't support it